File: solutions/erlang/raindrops/1/src/raindrops.erl
-module(raindrops).

-export([convert/1]).

convert(Number) -> 
    Words = lists:concat([convert(Number, Divisor) || Divisor <- [3, 5, 7]]),
    case Words of
        "" -> integer_to_list(Number);
        _ -> Words
    end.

convert(N, 3) when N rem 3 == 0 -> "Pling";
convert(N, 5) when N rem 5 == 0 -> "Plang";
convert(N, 7) when N rem 7 == 0 -> "Plong";
convert(_, _) -> "".

File: solutions/go/raindrops/1/raindrops.go
package raindrops

import "strconv"

// Convert returns string based on given rules applied to number input as specified:
// if number has 3 as a factor, add 'Pling' to the result
// if number has 5 as a factor, add 'Plang' to the result
// if number has 7 as a factor, add 'Plong' to the result
// if number does not have any of 3, 5, or 7 as a factor, the result should be the digits of the number
func Convert(number int) string {
	buffer := ""

	if number%3 == 0 {
		buffer += "Pling"
	}

	if number%5 == 0 {
		buffer += "Plang"
	}

	if number%7 == 0 {
		buffer += "Plong"
	}

	if buffer == "" {
		return strconv.Itoa(number)
	}

	return buffer
}
